refactor: applying minor caching of variables in a few files.

TheAvonian opened the pull request:
Very minor changes to assure no null accesses (even though it shouldn't be possible in these places) and caching `SocialData` in several methods where there is heavy usage of `PlayerInfo.get(SocialData.class)`.

DonkeyBlaster commented on src/main/java/com/wynntils/modules/core/events/ServerEvents.java:
```diff
@@ -142,15 +142,20 @@ public void chatMessage(ClientChatReceivedEvent e) {
         PartyManager.handleMessages(e.getMessage());  // party messages here
 
+        SocialData socialData = PlayerInfo.get(SocialData.class);
+        if(socialData == null) {
```
```suggestion
        if (socialData == null) {
```

DonkeyBlaster commented on src/main/java/com/wynntils/modules/core/events/ServerEvents.java:
```diff
@@ -98,14 +98,14 @@ public void joinServer(FMLNetworkEvent.ClientConnectedToServerEvent e) {
     @SubscribeEvent(priority = EventPriority.LOWEST)
     public void joinWorldEvent(WynnWorldEvent.Join e) {
-        if (PlayerInfo.get(CharacterData.class).getClassId() == -1 || CoreDBConfig.INSTANCE.lastClass == ClassType.NONE)
+        CharacterData data = PlayerInfo.get(CharacterData.class);
```
I think this would benefit from being called `characterData` instead of `data`

DonkeyBlaster commented on src/main/java/com/wynntils/modules/core/events/ServerEvents.java:
```diff
@@ -210,14 +215,19 @@ public void chatMessage(ClientChatReceivedEvent e) {
     @SubscribeEvent
     public void addFriend(ClientChatEvent e) {
+        SocialData socialData = PlayerInfo.get(SocialData.class);
+        if(socialData == null) {
```
```suggestion
        if (socialData == null) {
```

DonkeyBlaster commented on src/main/java/com/wynntils/modules/core/events/ServerEvents.java:
```diff
@@ -333,6 +343,10 @@ private static void startUpdateRegionName() {
             FrameworkManager.getEventBus().post(new SchedulerEvent.RegionUpdate());
 
             LocationData data = PlayerInfo.get(LocationData.class);
+            if(data == null) {
```
Fix spacing, also maybe call this `locationData`

DonkeyBlaster commented on src/main/java/com/wynntils/modules/core/managers/PartyManager.java:
```diff
@@ -20,15 +22,20 @@ public class PartyManager {
-        PlayerInfo.get(SocialData.class).resetPlayerParty();
+
+        SocialData socialData = PlayerInfo.get(SocialData.class);
+        if(socialData == null) {
```
```suggestion
        if (socialData == null) {
```

DonkeyBlaster commented on src/main/java/com/wynntils/modules/core/managers/PartyManager.java:
```diff
@@ -47,36 +54,46 @@ public static void handlePartyList() {
-        PartyContainer partyContainer = PlayerInfo.get(SocialData.class).getPlayerParty();
+
+        SocialData socialData = PlayerInfo.get(SocialData.class);
+        if(socialData == null) {
```
```suggestion
        if (socialData == null) {
```

DonkeyBlaster reviewed (commented):
A few things I noticed while skimming through this quickly. I will take another look in the morning or some other time

TheAvonian commented:
I fixed the spacing, I forgot to reformat before I committed at the time. Also changed out the generic data names into the more specific ones suggested.

kristofbolyai reviewed (approved):
This seems okay. Thanks.

kristofbolyai commented:
I am wondering though.. Was there a crash related to these being null? Where did the idea of making these specific methods null-safe come from?

TheAvonian commented:
The idea more so came from me working on another feature and realizing in a lot of files there is heavy use of `PlayerStats.get(T)` within the same method bodies so I went through those ones I was using and just made them try to call it once/twice at most in the body as every call to it is a foreach loop that is unneeded. 
Honestly the null checks are unneeded additions based on how everything is setup as it seems like they could never be null at those points but I added it for the sake of assurance. 

Ideally I would do this to every file but I only made these changes on my open files at the time.

kristofbolyai commented:
Yeah, this feels a bit paranoid. But then, the design of those "getters" are kinda bad imo.. So it should be fine to keep it.
